File: BW/10월2주차/JadenCase문자열만들기.js
function solution(s) {
    var str = s.toLowerCase().split('');
    if(isNaN(str[0])) str[0] = str[0].toUpperCase();
    
    for(var i=0; i<str.length-1; i++){
        if(str[i] === ' ' && isNaN(str[i+1])){
            str[++i] = str[i].toUpperCase();
            continue;
        }
    }
    
    if(str[str.length-2] === ' ' && isNaN(str[str.length-1])){
        str[str.length-1] = str[str.length].toUpperCase();
    }
    str = str.join('');
    
    return str;
}

File: MK/10월 2주차/JadenCase문자열만들기.java
class Solution {
    public String solution(String s) {

        StringBuilder answer = new StringBuilder();

        String[] arr = s.split(" ");

        for(int i = 0; i < arr.length; i++){
            if(arr[i].length() == 0){
                answer.append(" ");
            }
            else{
                answer.append(arr[i].substring(0, 1).toUpperCase())
                        .append(arr[i].substring(1).toLowerCase())
                        .append(" ");
            }
        }

        if(s.charAt(s.length()-1) == ' ')
            return answer.toString();

        answer.deleteCharAt(answer.length()-1);

        return answer.toString();
    }
}